pkg/pool-linear/contracts/euler/EulerLinearPoolFactory.sol

```diff
@@ -0,0 +1,152 @@
+        // `IVault.registerTokens` with it, and the Asset Manager must know about the Pool in order to store its Pool
+        // ID, wrapped and main tokens, etc., as immutable variables.
+        // We could forego immutable storage in the Rebalancer and simply have a two-step initialization process that
+        // uses storage, but we can keep those gas savings by instead making the deployment a bit more complicated.
+        //
+        // Note that the Pool does not interact with the Asset Manager: it only needs to know about its address.
+        // We therefore use create2 to deploy the Asset Manager, first computing the address where it will be deployed.
+        // With that knowledge, we can then create the Pool, and finally the Asset Manager. The only issue with this
+        // approach is that create2 requires the full creation code, including constructor arguments, and among those is
+        // the Pool's address. To work around this, we have the Rebalancer fetch this address from `getLastCreatedPool`,
+        // which will hold the Pool's address after we call `_create`.
+
+        bytes32 rebalancerSalt = bytes32(_nextRebalancerSalt);
+        _nextRebalancerSalt += 1;
+
+        bytes memory rebalancerCreationCode = abi.encodePacked(
+            type(EulerLinearPoolRebalancer).creationCode,
+            abi.encode(getVault(), _queries, EULER_PROTOCOL)
+        );
+        address expectedRebalancerAddress = Create2.computeAddress(rebalancerSalt, keccak256(rebalancerCreationCode));
+
+        (uint256 pauseWindowDuration, uint256 bufferPeriodDuration) = getPauseConfiguration();
+
+        EulerLinearPool.ConstructorArgs memory args = EulerLinearPool.ConstructorArgs({
+            vault: getVault(),
+            name: name,
+            symbol: symbol,
+            mainToken: mainToken,
+            wrappedToken: wrappedToken,
+            assetManager: expectedRebalancerAddress,
+            upperTarget: upperTarget,
+            swapFeePercentage: swapFeePercentage,
+            pauseWindowDuration: pauseWindowDuration,
+            bufferPeriodDuration: bufferPeriodDuration,
+            owner: owner,
+            version: getPoolVersion()
+        });
+
+        EulerLinearPool pool = EulerLinearPool(_create(abi.encode(args)));
+
+        // LinearPools have a separate post-construction initialization step: we perform it here to
+        // ensure deployment and initialization are atomic.
+        pool.initialize();
+
+        // Not that the Linear Pool's deployment is complete, we can deploy the Rebalancer, verifying that we correctly
+        // predicted its deployment address.
+        address actualRebalancerAddress = Create2.deploy(0, rebalancerSalt, rebalancerCreationCode);
+        require(expectedRebalancerAddress == actualRebalancerAddress, "Rebalancer deployment failed");
+
+        require(
+            EULER_PROTOCOL == IEulerLinearPoolRebalancer(actualRebalancerAddress).EULER_PROTOCOL(),
```

Comment:
@rabmarut the reason the Interface mentioned in https://github.com/mkflow27/balancer-v2-monorepo/pull/2#discussion_r1042286641 was added is due to this additional require statement I added.

Reply:
Ah, I didn't see it. Thank you!

By the way, you can reply directly to my comment above instead of making a new one.